frontend/src/app/core/helpers/exclusion-date-helper.ts

```diff
@@ -0,0 +1,150 @@
+import { DateAdapter } from '@angular/material/core';
+import { DayAction } from '@core/enums/day-action.enum';
+import { FindSameExclusionDateHelper } from '@core/helpers/find-same-exclusion-date-helper';
+import { convertTimeToOffset, getTimeString } from '@core/helpers/time-helper';
+import { TimeRangesMergeHelper } from '@core/helpers/time-ranges-merge-helper';
+import { getDateWithoutLocalOffset } from '@core/helpers/time-zone-helper';
+import { IDayTimeRange } from '@core/models/schedule/exclusion-date/IDayTimeRange';
+import { IExclusionDate } from '@core/models/schedule/exclusion-date/IExclusionDate';
+
+export const getExclusionDate = (
+    selected: string,
+    startHours: number,
+    startMinutes: string,
+    endHours: number,
+    endMinutes: string,
+): IExclusionDate => ({
+    selectedDate: selected,
+    dayTimeRanges: [
+        {
+            start: getTimeString(startHours, startMinutes),
+            end: getTimeString(endHours, endMinutes),
+        },
+    ],
+});
+
+export function getExclusionDateWithDayOffset(
+    exclusionDate: IExclusionDate,
+    dateAdapter: DateAdapter<Date>,
+    startHours: number,
+    startMinutes: string,
+    endHours: number,
+    endMinutes: string,
+    dayAction: DayAction,
+) {
+    let convertedDate: Date;
+
+    if (dayAction === DayAction.AddDay) {
+        convertedDate = dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), 1);
+    } else if (dayAction === DayAction.SubtractDay) {
+        convertedDate = dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), -1);
+    } else {
+        convertedDate = new Date(exclusionDate.selectedDate);
+    }
+
+    return [
+        getExclusionDate(
+            getDateWithoutLocalOffset(convertedDate).toJSON(),
+            startHours,
+            startMinutes,
+            endHours,
+            endMinutes,
+        ),
+    ];
+}
+
+export const getConvertedDatesForExclusionDate = (
+    exclusionDate: IExclusionDate,
+    dateAdapter: DateAdapter<Date>,
+    startHoursDayAction: DayAction,
+): [string, string] =>
+    (startHoursDayAction === DayAction.SubtractDay
+        ? [
+            getDateWithoutLocalOffset(dateAdapter.addCalendarDays(new Date(exclusionDate.selectedDate), -1)).toJSON(),
```

Comment:
Why do we need JSON?

Reply:
In order to convert date to json string and then pass this string in request. Native date object during convertation to json is converted with user offset from his PC. To avoid this I convert it manually and then pass string, when I need Date object, I create it from string right in place

Reply:
Fixed it by moving toJSON() to helper method